strax/context.py

```diff
@@ -239,6 +239,39 @@ def register(self, plugin_class):
         for p in plugin_class.provides:
             self._plugin_class_registry[p] = plugin_class
 
+        already_seen = []
```

Comment:
- Why would you expect to see plugins multiple times? Even if you would, you make a new list each time you call ```register``` so I'm not sure if it'd help.

Reply:
Jelle just answered this question. Your way of keeping track of what has been seen is somewhat faster indeed.